pkg/agent/cniserver/server_windows_test.go

```diff
@@ -0,0 +1,1022 @@
+}
+
+func (t *hnsTestUtil) attachEndpointInNamespace(ep *hcn.HostComputeEndpoint, namespace string) error {
+	t.hcnEndpoint.HostComputeNamespace = namespace
+	if t.endpointAttachErr == nil {
+		t.addHostInterface()
+	}
+	return t.endpointAttachErr
+}
+
+func (t *hnsTestUtil) removeEndpointFromNamespace(namespace string, epID string) error {
+	return nil
+}
+
+func (t *hnsTestUtil) setFunctions() {
+	listHnsEndpointFunc = t.listHnsEndpointFunc
+	createHnsEndpointFunc = t.createHnsEndpoint
+	getNamespaceEndpointIDsFunc = t.getNamespaceEndpointIDs
+	hotAttachEndpointFunc = t.hotAttachEndpoint
+	attachEndpointInNamespaceFunc = t.attachEndpointInNamespace
+	isContainerAttachOnEndpointFunc = t.isContainerAttachOnEndpoint
+	getHcnEndpointByIDFunc = t.getHcnEndpointByID
+	deleteHnsEndpointFunc = t.deleteHnsEndpoint
+	removeEndpointFromNamespaceFunc = t.removeEndpointFromNamespace
+}
+
+func (t *hnsTestUtil) deleteHostIface(ifaceName string) {
+	hostIfaces.Delete(ifaceName)
+}
+
+func (t *hnsTestUtil) addHostInterface() {
+	if _, exists := hostIfaces.Load(t.hostIfaceName); exists {
+		return
+	}
+	go func() {
+		select {
+		case <-time.After(time.Millisecond * 650):
+			hostIfaces.Store(t.hostIfaceName, true)
+		}
+	}()
+}
+
+func TestCmdAdd(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	ipamType := "windows-test"
+	ipamMock := ipamtest.NewMockIPAMDriver(controller)
+	ipam.ResetIPAMDriver(ipamType, ipamMock)
+	mockOVSBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(controller)
+	mockOFClient := openflowtest.NewMockClient(controller)
+	routeMock := routetest.NewMockInterface(controller)
+	var ifaceStore interfacestore.InterfaceStore
+	ipamResult := &ipam.IPAMResult{Result: *ipamtest.GenerateIPAMResult(cni.AntreaCNIVersion, ips, routes, dns)}
+	ctx := context.TODO()
+
+	newServer := func(podUpdateNotifier *channel.SubscribableChannel) *CNIServer {
+		ifaceStore = interfacestore.NewInterfaceStore()
+		cniServer := newCNIServer(t)
+		cniServer.routeClient = routeMock
+		gwMAC, _ := net.ParseMAC("00:00:11:11:11:11")
+		cniServer.podConfigurator, _ = newPodConfigurator(mockOVSBridgeClient, mockOFClient, routeMock, ifaceStore, gwMAC, "system", false, podUpdateNotifier, nil, false)
+		return cniServer
+	}
+
+	prepareSetup := func(name string, containerID, infraContainerID, netns string, withPreviousResult bool) (*cnipb.CniCmdRequest, string) {
+		networkCfg := generateNetworkConfiguration("", supportedCNIVersion, "", ipamType)
+		if withPreviousResult {
+			networkCfg.RawPrevResult, _ = translateRawPrevResult(&ipamResult.Result, supportedCNIVersion)
+		}
+		podArgs := cniservertest.GenerateCNIArgs(name, testPodNamespace, containerID)
+		requestMsg, _ := newRequest(podArgs, networkCfg, "", t)
+		requestMsg.CniArgs.ContainerId = containerID
+		requestMsg.CniArgs.Netns = netns
+		hostIfaceName := util.GenerateContainerInterfaceName(name, testPodNamespace, infraContainerID)
+		return requestMsg, hostIfaceName
+	}
+
```

Comment:
Could you check if `prepareSetup` can be merged into a general function?  I didn't see much difference in three `prepareSetup`.